KarhooUISDK/Screens/CheckoutScreen/Components/BottomSheets/FlightNumber/KarhooBottomSheetContentWithTextFieldViewModel.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  KarhooBottomSheetContentWithTextFieldViewModel.swift
+//  KarhooUISDK
+//
+//  Created by Bartlomiej Sopala on 16/01/2023.
+//  Copyright © 2023 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+
+final class KarhooBottomSheetContentWithTextFieldViewModel: ObservableObject {
+    @Published var textFieldText: String
+    @Published var isTextfieldValid: Bool
+
+    var initialValueForTextField: String
+    var errorMessage: String
+    var onSaveCallback: (_ newFlightNumber: String) -> Void
+
+    let viewSubtitle: String
+    let buttonText = UITexts.Generic.save
+    let textFieldHint: String
+    
+    init(
+        initialValueForTextField: String,
+        viewSubtitle: String,
+        textFieldHint: String,
+        errorMessage: String,
+        onSaveCallback:  @escaping (_ newFlightNumber: String) -> Void
+    ) {
+        self.initialValueForTextField = initialValueForTextField
+        self.viewSubtitle = viewSubtitle
+        self.textFieldHint = textFieldHint
+        self.onSaveCallback = onSaveCallback
+        self.errorMessage = errorMessage
+        textFieldText = initialValueForTextField
+        isTextfieldValid = true
+    }
+    
+    func didTapSave(textFieldValue: String){
+        onSaveCallback(textFieldValue)
```

Comment:
We could think about replacing simple callbacks with Combine (https://developer.apple.com/documentation/combine/using-combine-for-your-app-s-asynchronous-code). This way, we'd have a practice once the time comes for more complex solutions. 

It's quite simple tbh:

```
final class KarhooBottomSheetContentWithTextFieldViewModel: ObservableObject {
    @Published var textFieldText: String
    @Published var isTextfieldValid: Bool

    var initialValueForTextField: String
    var errorMessage: String
    var onSaveSubject:  PassthroughSubject<String, Never>

    let viewSubtitle: String
    let buttonText = UITexts.Generic.save
    let textFieldHint: String

    init(
        initialValueForTextField: String,
        viewSubtitle: String,
        textFieldHint: String,
        errorMessage: String,
        onSaveSubject:  PassthroughSubject<String, Never>
    ) {
        self.initialValueForTextField = initialValueForTextField
        self.viewSubtitle = viewSubtitle
        self.textFieldHint = textFieldHint
        self.onSaveSubject = onSaveSubject
        self.errorMessage = errorMessage
        textFieldText = initialValueForTextField
        isTextfieldValid = true
    }

    func didTapSave(textFieldValue: String){
        onSaveSubject.send(textFieldValue)
    }
}
```

while observer needs to implement something like:
```
onSaveSubject.eraseToAnyPublisher()
    .sink { newTextfieldValue in 
        // handle new value
    }
    .store(in: &cancelables)
```

Reply:
It'd require some additional work to rework, so maybe don't do it now, since the code works great, but let's think about this for next features, maybe?